.github/workflows/lint.yml

```diff
@@ -0,0 +1,43 @@
+name: Lint
+
+on: push
+
+jobs:
+  run-linters:
+    name: Google Java Format
+    runs-on: ubuntu-latest
+
+    steps:
+      - name: Check out Git repository
+        uses: actions/checkout@v2 # v2 minimum required
+
+      - name: Set up JDK 11
+        uses: actions/setup-java@v1
+        with:
+          java-version: "11" # Recommended: latest versions of Google Java Format require JDK 11+
+
+      - name: Set up Node.js
+        uses: actions/setup-node@v1
+        with:
+          node-version: 14
+
+      # ESLint must be in `package.json`
+      - name: Install Node.js dependencies
+        run: npm install
+        working-directory: ./frontend
+
+      - name: Run Google Java Format
+        uses: axel-op/googlejavaformat-action@v3
+        with:
+          skipCommit: "true"
+          args: "--skip-sorting-imports --dry-run --set-exit-if-changed"
```

Comment:
I couldn't find what these flags do in their documentation, I assume it's so that it doesn't actually change the code? Or what?

Reply:
Here is the [documentation](https://github.com/axel-op/googlejavaformat-action) on the flags. Correct, we run the format without committing the changes automatically. Instead, we get a failed check with the names of the files that need formatting.

Also regarding --skip-sorting-imports, IntelliJ requires a small extra configuration step for import sorting and it's possible that different versions may not sort imports exactly the same, so for check purposes it's slightly unclear what import sort the GitHub Action enforces versus that of other plugins. Basically import sorting is skipped to avoid possible head scratching moments. However, the check will still check if there are unused imports to remove.